test/langtools/jdk/internal/shellsupport/doc/SinceCheckerHelper.java

```diff
@@ -0,0 +1,317 @@
+import com.sun.source.util.JavacTask;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.code.Symbol;
+import jdk.internal.shellsupport.doc.JavadocHelper;
+
+import javax.lang.model.element.*;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.*;
+import java.io.IOException;
+import java.lang.Runtime.Version;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class SinceCheckerHelper {
```

Comment:
A bit general comment - I would separate the "fill the database" and "check the tags" in the source - methods that belong to one of these should be close to other methods in the same category, and ideally the groups should be separated with comments, both between themselves, and between the groups and other general code in the class.